config/src/config.rs

```diff
@@ -398,4 +418,45 @@ mod tests {
             test_value
         );
     }
+
+    // Test to check substitute_env_vars function is substituting env variables
+    // indexers can use ${ENV_VAR_NAME} to point to the required env variable
+    #[test]
+    fn test_substitute_env_vars() {
+        // Set up environment variables
+        env::set_var("TEST_VAR1", "hello");
+        env::set_var("TEST_VAR2", "world");
+        // TEST_VAR3 not setup and should not be replaced by the function
+
+        let input = r#"
+            [section1]
+            key1 = "${TEST_VAR1}"
+            key2 = "${TEST_VAR2}"
```

Comment:
Could you test some other keys and edge cases?

Like just "{TEST_VAR}" (without the $) or "${{TEST_VAR}}".

Some other cases like "${TEST_VAR:-default}" and "${TEST_VAR-default}". 

Just to have a clear understanding how the replacement works (you don't need to implement default values).

Reply:
got it[✌️]

Reply:
Hey, could you also set TEST_VAR to a value or update it to TEST_VAR1? Sorry I wasn't clear on my previous comment.

Reply:
still a little confused[🥲], did you mean ${TEST_VAR:-default} to a default value, or assign a value to it, as of it represents an variable which is not available in .env and wont be replaced

Reply:
if the corresponding env variables are not set, it should send an error warning to user right? rather than replacing it with default values

Reply:
sorry for the confusion hahaha. I told in the comment to use TEST_VAR, but I want to understand what happens when there are default values and the env var is set. So I should have asked for TEAT_VAR1 (which is already defined)

Reply:
cool, I'll setup a modify test, so replacement will be checked